scripts/pylib/twister/twisterlib.py

```diff
@@ -896,6 +896,16 @@ def handle(self):
 
         ser.flush()
 
+        # turns out the ser.flush() is not enough to clear serial leftover from last case
+        # explicitly readline() can do it reliably
+        old_timeout = ser.timeout
+        ser.timeout = 1
+        leftover = ser.readline()
+        while leftover:
+            logger.debug(f"leftover log of previous test: {leftover}")
+            leftover = ser.readline()
+        ser.timeout = old_timeout
+
```

Comment:
what if there is a dead loop console output?

Reply:
If there's no leftover serial output, the `ser.readline()` will timeout and return `b''`, which will end the loop. 
If there's some endless output from previous test case, it is the test case that should be fixed.
So I think it is OK to do a simple processing like this.

Reply:
the problem is that if last test case fails and prints endless output, then it will make twister in deadloop, actually, after flash the new application, the last case will not impact current case. I guess we do not want the test blocked by a single failure.

Reply:
I think your point is a new flash can clear the effect of last bad case. But actually it can't. 
The serial port monitoring thread is opened before the flash action (https://github.com/zephyrproject-rtos/zephyr/blob/f875d027a04374fcf07661809f81a42c2c646629/scripts/pylib/twister/twisterlib.py#L908), which is to avoid missing any new logs from the new image. But unfortunately this early thread will catch the leftover logs from last case if any. So I think it is necessary to drain the serial port before a the serial monitoring thread starts.

And I think no test case should produce endless logs. If it does, the case must be fixed.

Reply:
@smrtos , no case will create endless log this is not a true statement. When cases goes wrong, such thing will happen. It may not be the case issue, but some defects in driver or zephyr kernel. if such thing happens, do we want to accept that all following cases will be blocked?

Reply:
Just refined the code. Could you take another look? @hakehuang